enhancements/installer/aws-custom-edge-machineset-local-zones.md

```diff
@@ -0,0 +1,933 @@
+
+> TODO: check if the current deployment is not exposing any port to the `0.0.0.0/0` on `worker` SG
+
+#### Infrastructure costs
+
+The internal data transfer costs (within the VPC: kubelet<->NLBs) will be the same
+of the other (normal workers) nodes[1][2].
+
+> *Data Transfer charges in AWS Local Zones is the same as in the Availability Zones today. Learn more about data transfer prices "in" and "out" of Amazon EC2 here.*
+
+The compute resource in Local Zones is more expensive, for example, `m5.xlarge` hosted
+in the parent zones of `us-west-2` will cost `$0.192/hr`, and in the LAX (Los Angeles)
+Local Zones is `$0.23/hr`, almost 17% more expensive than the regular zone.
+
+When following the architecture of using private subnets in Local Zones re-using Nat Gateway
+from the parent region (proposal #2 in the last section), considering NAT Gateway
+currently is not supported on the Local Zones locations, the following fees will be
+applied when the node needs access to the internet:
+
+- Fee for data transferred "in" to and "out" from NAT GW located in another AZ (`$0.01/GB` for each direction)
+- NAT GW fee per GB processed `$0.045`
+
+There is another cost of `$0.045/hr` for NAT GW allocation, which is not exclusive to a node in
+Local Zones as the Nat Gateway is already created to be used in a regular zone.
+
+When the node is in the public subnet (described in this EP), the fees applied will be the same to access the internet.
+
+[1] [EC2 Data transfer fees](https://aws.amazon.com/ec2/pricing/on-demand/)
+[2] [EC2 Local Zones Pricing](https://aws.amazon.com/about-aws/global-infrastructure/localzones/pricing/):
+
+
+#### User-workload deployments
+
+Application developers must set the tolerations for the application running for each location.
+
+The pod spec example below describes how to deploy one application on the node on the zone group `ZONE_GROUP`:
+
+```yaml
+      nodeSelector:
+        zone_group: ZONE_GROUP
+      tolerations:
+      - key: "node-role.kubernetes.io/edge"
+        operator: "Equal"
+        value: ""
+        effect: "NoSchedule"
+```
+
+<!--
+> TODO create one example of deploying the application on the same ZONE_GROUP, spreading it into different zones.
+
+> Currently only Los Angeles (LAX) location has more than one zone[1];
+
+The example below shows how to deploy one application with two replicas distributed
+between two Local Zones[1] in the same zone group[2].
+
+[1] Local Zones available in Los Angeles `us-west-2-lax-1a` and `us-west-2-lax-1b`
+[2] Local Zones group of Los Angeles `us-west-2-lax-1` -->
+
+
+#### User-workload ingress traffic
+
+To expose the applications to the internet on AWS Local Zones, application developers
+must expose their applications using AWS Application Load Balancers (ALB). The
+[ALB Operator](https://docs.openshift.com/container-platform/4.11/networking/aws_load_balancer_operator/install-aws-load-balancer-operator.html) is available through OLM on 4.11+.
```

Comment:
Does the service object not support ALB when implemented by the AWS cloud provider?

Reply:
Yeah, AFAIK the AWS cloud provider supports only network load balancers (CLB/NLB) natively. ALB is handled by [AWS Load Balancer Controler](https://kubernetes-sigs.github.io/aws-load-balancer-controller).